subsys/rtio/rtio_executor.c

```diff
@@ -20,11 +20,17 @@ LOG_MODULE_REGISTER(rtio_executor, CONFIG_RTIO_LOG_LEVEL);
  */
 static inline void rtio_iodev_submit(struct rtio_iodev_sqe *iodev_sqe)
 {
+	if (FIELD_GET(RTIO_SQE_CANCELED, iodev_sqe->sqe.flags)) {
+		/* Canceled */
+		while (iodev_sqe != NULL) {
+			rtio_iodev_sqe_err(iodev_sqe, -ECANCELED);
+			iodev_sqe = iodev_sqe->next;
```

Comment:
There's a next inline available for this I'd prefer to see used here to avoid directly poking at the next member

Reply:
Done